### PR TITLE
Do no show empty reminder/rss tabs on home page

### PR DESCRIPTION
**Pull request by AdrienClairembault**
## Checklist before requesting a review

- [X] I have performed a self-review of my code.
- [X] I have added tests that prove my fix is effective or that my feature works.

## Description

On the helpdesk home page, do not display the `Reminder` and `Rss feeds` tabs if they have no content.

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: src/Glpi/Helpdesk/HomePageTabs.php [hunk 1]
```diff
@@ -80,15 +80,21 @@ public function getTabNameForItem(
             ),
         ];
 
-        if (Session::haveRight("reminder_public", READ)) {
+        if (
+            Session::haveRight("reminder_public", READ)
+            && Reminder::countPublicReminders() > 0
+        ) {
             // TODO: only show if at least one reminder is visible
```
Comment (cedric-anne) [inline src/Glpi/Helpdesk/HomePageTabs.php:87]:
```suggestion
```

File: src/Glpi/Helpdesk/HomePageTabs.php [hunk 2]
```diff
@@ -80,15 +80,21 @@ public function getTabNameForItem(
...
                 icon: Reminder::getIcon()
             );
         }
 
-        if (Session::haveRight("rssfeed_public", READ)) {
+        if (
+            Session::haveRight("rssfeed_public", READ)
+            && RSSFeed::countPublicRssFedds() > 0
+        ) {
             // TODO: only show if at least one RSS feed is visible
```
Comment (cedric-anne) [inline src/Glpi/Helpdesk/HomePageTabs.php:98]:
```suggestion
```

File: src/RSSFeed.php [hunk 3]
```diff
@@ -486,21 +486,30 @@ public function prepareInputForAdd($input)
...
 
-        if ($feed = self::getRSSFeed($input['url'])) {
-            $input['have_error'] = 0;
-            $input['name']       = $feed->get_title();
-            if (empty($input['comment'])) {
-                $input['comment'] = $feed->get_description();
+        // We may want to disable the name computation when working with fake
+        // feeds in our unit tests
+        $compute_name = $input['_do_not_compute_name'] ?? false;
+        if (!$compute_name) {
```
Comment (cedric-anne) [inline src/RSSFeed.php:492]:
```suggestion
        // We may want to disable the title/description values fetching when working with fake
        // feeds in our unit tests
        $fetch_values = ($input['_do_not_fetch_values'] ?? false) === false;
        if ($fetch_values) {
```

File: src/RSSFeed.php [hunk 4]
```diff
@@ -486,21 +486,30 @@ public function prepareInputForAdd($input)
...
         }
         $input["name"] = trim($input["name"]);
 
         if (empty($input["name"])) {
             $input["name"] = __('Without title');
         }
+
+        // Owner cannot be changed
+        unset($input['users_id']);
+
```
Comment (cedric-anne) [inline src/RSSFeed.php:512]: This value is forced on line 487.
```suggestion
```

File: phpunit/functional/Glpi/Helpdesk/HomePageTabsTest.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+            'password2'     => 'tmp_user',
+            '_profiles_id'  => getItemByTypeName(Profile::class, 'Self-Service', true),
+            '_entities_id'  => $this->getTestRootEntity(true),
+            '_is_recursive' => true,
+        ], ['password', 'password2']);
+        $this->login('tmp_user', 'tmp_user'); // Need to be logged in to create the RSS feed
+        $reminder = $this->createItem(RSSFeed::class, [
+            'name' => 'My feed',
+            'url'  => 'https://fake-rss.localhost.com/feed',
+            '_do_not_compute_name' => true,
```
Comment (cedric-anne) [inline phpunit/functional/Glpi/Helpdesk/HomePageTabsTest.php:117]:
```suggestion
            '_do_not_fetch_values' => true,
```

File: src/Reminder.php [hunk 6]
```diff
@@ -667,6 +652,54 @@ public static function showListForCentral(bool $personal = true, bool $display =
             $public_criteria   = array_merge_recursive($public_criteria, $additional_criteria);
         }
 
+        // TOOD: would be cleaner to have two separate method like getPersonalCriteria
+        // and getPublicCriteria
```
Comment (cedric-anne) [inline src/Reminder.php:656]: You can do it by passing a `bool $personal` parameter as you did for RSS feeds.

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/RSSFeed.php (hunk 4 above)
Comment (AdrienClairembault) [reply src/RSSFeed.php:512]: Indeed, it comes from a bad rebase I think.

File: src/Reminder.php (hunk 6 above)
Comment (AdrienClairembault) [reply src/Reminder.php:656]: I don't want to do it now, I've just moved the original code without too much modification as it is not the goal of this PR nor is it something high priority (and there are a lot others problems with this code that need to be fixed).

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**